lib/chef/cookbook/metadata.rb

```diff
@@ -612,8 +650,28 @@ def privacy(arg=nil)
         )
       end
 
+      def validate_ohai_version!
+        unless gem_dep_matches?("ohai", Gem::Version.new(Ohai::VERSION), *ohai_versions)
+          raise Exceptions::CookbookOhaiVersionMismatch.new(Ohai::VERSION, name, version, *ohai_versions)
+        end
+      end
+
+      def validate_chef_version!
+        unless gem_dep_matches?("chef", Gem::Version.new(Chef::VERSION), *chef_versions)
+          raise Exceptions::CookbookChefVersionMismatch.new(Chef::VERSION, name, version, *chef_versions)
+        end
+      end
+
     private
 
+      def gem_dep_matches?(what, version, *deps)
+        return true unless deps.length > 0
+        deps.each do |dep|
```

Comment:
Ahh, I think I see how this works. Each "row" is loaded into a single dependency object so that covers the `all?`, and then we want an `any?` between each of those? That seems maybe a bit overcomplicated but I can dig it. I would still rewrite this as an `any?` for clarity.